src/pkcs7.rs

```diff
@@ -1,20 +1,43 @@
-
 pub mod pkcs7 {
+    use std::fmt;
+    use std::error::Error;
+
+    type Result<T> = std::result::Result<T, Pkcs7UnpadError>;
+
+    #[derive(Debug)]
+    pub struct Pkcs7UnpadError;
+
+    impl fmt::Display for Pkcs7UnpadError {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            write!(f, "invalid pkcs7 padding")
+        }
+    }
+
+    impl Error for Pkcs7UnpadError {
+        fn description(&self) -> &str {
+            "PKCS7 unpadding error"
+        }
+    }
+
     pub fn pad(input: &[u8], key_size: usize) -> Vec<u8> {
         let pad_num: usize;
-        if input.len() == key_size {
-            pad_num = key_size;
-        } else if input.len() < key_size {
-            pad_num = key_size - input.len();
-        } else {
-            pad_num = key_size - (input.len() % key_size);
-        }
+        pad_num = key_size - (input.len() % key_size);
         [input.to_vec(), vec![pad_num as u8; pad_num]].concat()
     }
     
-    pub fn unpad(input: &[u8]) -> Vec<u8> {
+    pub fn unpad(input: &[u8], key_size: usize) -> Result<Vec<u8>> {
         let pad_num = input.last().unwrap();
```

Comment:
Seems like it might be convenient to cast as `usize` here instead of repeatedly below?